### PR TITLE
Add doc_str and doc macros, to support Base.Docs/Docile syntax.

### LINKED ISSUE #56: make `@doc` a no-op if Docile can't be loaded?
I've had some trouble getting Docile installed on some older versions of Julia. Since Docile provides no necessary functionality for most packages that "depend" on it, it would be nice to make this an optional dependency instead: if the Docile package cannot be loaded then just ignore `@doc` macros.

### PR DESCRIPTION
**Pull request by hayd**
fix #56.

### CODE REVIEW

File: src/Compat.jl [hunk 1]
```diff
@@ -38,6 +38,16 @@ if VERSION < v"0.4.0-dev+412"
     eval(Base, :(const IPAddr = IpAddr))
 end
 
+if VERSION < v"0.4.0-dev"
+    try
+        import Docile: @doc_str, @docmstr, @doc
```
Comment (MichaelHatherly) [inline src/Compat.jl]: ``@docmstr`` -> ``@doc_mstr``

**Comment by MichaelHatherly** [comment]
The version condition should probably be a bit more specific since ``@doc`` didn't get added until later during 0.4.

File: src/Compat.jl (hunk 1 above)
Comment (hayd) [inline src/Compat.jl]: fixed, thanks!

**Comment by hayd** [comment]
Hmmm, it's been a long time > 6 months I think? Happy to tweak this if it's easy to work out.

**Comment by MichaelHatherly** [comment]
``0.4.0-dev+1501`` I think.

**Comment by hayd** [comment]
okey dokey, let's go with that. :)

**Comment by stevengj** [comment]
Needs a test.

**Comment by hayd** [comment]
Added test. locally this works on 0.3 both with and without Docile.

**Comment by hayd** [comment]
@MichaelHatherly actually when I look at this, help doesn't seem to work for me on 0.3 (before this PR). I was under the impression that

    help> func # would not show doc
    julia> using Docile
    help> func  # would show doc

Maybe there is something funky about my 0.3 install.

Should this work?

**Comment by MichaelHatherly** [comment]
> Should this work?

To get help in 0.3 for packages you need to import Lexicon first. Docile only provides the docstrings while Lexicon hooks into the base help system.

**Comment by hayd** [comment]
I think it would be nice to merge this soon...

One potential issue is this *may* cause some warnings where libraries have already tried to import Docile ? The trick being, remove the using/import of Docile and let Compat do the work. ?
(Perhaps this could be solved by moving the export into the catch block?)

**Comment by hayd** [comment]
I'm going to pull the trigger on this in the next few days unless there's any more comments.

**Comment by MichaelHatherly** [comment]
@hayd, this is causing failures for packages on 0.3 that import Docile *after* Compat, https://github.com/JuliaGraphics/ColorTypes.jl/blob/a8dd9a3834b76e48db138141ab586325cd4efaa6/src/ColorTypes.jl#L5-L8.

This probably needs to be reverted unless there's a way to avoid the imports clashing.

First see here https://groups.google.com/forum/#!topic/julia-users/rsM4hxdkAxg.

**Pull request merged**